extendedset/src/main/java/io/druid/extendedset/intset/EmptyIntIterator.java

```diff
@@ -17,47 +17,48 @@
  * under the License.
  */
 
-package io.druid.segment;
+package io.druid.extendedset.intset;
 
-import io.druid.segment.data.IndexedInts;
-import io.druid.segment.data.Offset;
+import java.util.NoSuchElementException;
 
-/**
-*/
-class IndexedIntsOffset implements Offset
+public final class EmptyIntIterator implements IntSet.IntIterator
 {
-  int currRow;
-  private final IndexedInts invertedIndex;
+  private static final EmptyIntIterator INSTANCE = new EmptyIntIterator();
 
-  public IndexedIntsOffset(IndexedInts invertedIndex)
+  public static EmptyIntIterator instance()
   {
-    this.invertedIndex = invertedIndex;
-    currRow = 0;
+    return INSTANCE;
+  }
+
+  private EmptyIntIterator() {}
+
+  @Override
+  public boolean hasNext()
+  {
+    return false;
   }
 
   @Override
-  public void increment()
+  public int next()
   {
-    ++currRow;
+    throw new NoSuchElementException();
   }
 
   @Override
-  public boolean withinBounds()
+  public void remove()
   {
-    return currRow < invertedIndex.size();
+    throw new UnsupportedOperationException();
   }
 
   @Override
-  public Offset clone()
+  public void skipAllBefore(int element)
   {
-    final IndexedIntsOffset retVal = new IndexedIntsOffset(invertedIndex);
-    retVal.currRow = currRow;
-    return retVal;
+    // nothing to skip
   }
 
   @Override
-  public int getOffset()
+  public IntSet.IntIterator clone()
   {
-    return invertedIndex.get(currRow);
+    return new EmptyIntIterator();
```

Comment:
I left it because `Object.clone()` contract implies that a new object is returned. I think this is not very harmful. This could be fixed later, if `IntIterator` is refactored to have a custom `copy()` method instead of inheriting `Object.clone()`.